### PR TITLE
Use allow list over whitelist

### PR DESCRIPTION
**Pull request by evan2645**
This commit updates all docs and code to remove the use of the term
whitelist in favor of allow list. Where necessary, deprecation warnings
are now logged.

Signed-off-by: Evan Gilman <egilman@vmware.com>

### CODE REVIEW

File: pkg/server/plugin/nodeattestor/gcp/iit.go [hunk 1]
```diff
@@ -95,15 +105,15 @@ func (p *IITAttestorPlugin) Attest(stream nodeattestorv0.NodeAttestor_AttestServ
...
 		if identityMetadata.ProjectID == projectID {
-			projectIDMatchesWhitelist = true
+			projectIDMatchesAllowList = true
 			break
 		}
 	}
-	if !projectIDMatchesWhitelist {
-		return pluginErr.New("identity token project ID %q is not in the whitelist", identityMetadata.ProjectID)
+	if !projectIDMatchesAllowList {
+		return pluginErr.New("identity token project ID %q is not in the allow_list", identityMetadata.ProjectID)
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/gcp/iit.go:116]:
```suggestion
		return pluginErr.New("identity token project ID %q is not in the allow list", identityMetadata.ProjectID)
```

File: pkg/server/plugin/nodeattestor/gcp/iit_test.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ func (s *IITAttestorSuite) TestErrorOnProjectIdMismatch() {
 		Data: s.signToken(token),
 	}
 	_, err := s.attest(&nodeattestorv0.AttestRequest{AttestationData: data})
-	s.RequireErrorContains(err, `gcp-iit: identity token project ID "project-whatever" is not in the whitelist`)
+	s.RequireErrorContains(err, `gcp-iit: identity token project ID "project-whatever" is not in the allow_list`)
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/gcp/iit_test.go:159]:
```suggestion
	s.RequireErrorContains(err, `gcp-iit: identity token project ID "project-whatever" is not in the allow list`)
```

File: pkg/server/plugin/nodeattestor/gcp/iit_test.go [hunk 3]
```diff
@@ -375,31 +375,31 @@ func (s *IITAttestorSuite) TestConfigure() {
...
 		Configuration: `
-projectid_whitelist = ["bar"]
+projectid_allow_list = ["bar"]
 `,
 		GlobalConfig: &plugin.ConfigureRequest_GlobalConfig{}})
 	s.RequireErrorContains(err, "gcp-iit: trust_domain is required")
 	require.Nil(resp)
 
-	// missing projectID whitelist
+	// missing projectID allow_list
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/gcp/iit_test.go:392]:
```suggestion
	// missing projectID allow list
```

File: pkg/server/plugin/nodeattestor/k8s/psat/psat_test.go [hunk 4]
```diff
@@ -391,14 +391,14 @@ func (s *AttestorSuite) TestConfigure() {
 	s.RequireGRPCStatus(err, codes.Unknown, "k8s-psat: configuration must have at least one cluster")
 	s.Require().Nil(resp)
 
-	// cluster missing service account whitelist
+	// cluster missing service account allow_list
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/k8s/psat/psat_test.go:394]:
```suggestion
	// cluster missing service account allow list
```

File: pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go [hunk 5]
```diff
@@ -329,7 +329,7 @@ func (s *AttestorSuite) TestConfigure() {
 	s.RequireGRPCStatus(err, codes.Unknown, "k8s-sat: configuration must have at least one cluster")
 	s.Require().Nil(resp)
 
-	// cluster missing service account whitelist (local validation config)
+	// cluster missing service account allow_list (local validation config)
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go:332]:
```suggestion
	// cluster missing service account allow list (local validation config)
```

File: pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go [hunk 6]
```diff
@@ -338,10 +338,10 @@ func (s *AttestorSuite) TestConfigure() {
 		}`, s.fooCertPath()),
 		GlobalConfig: &plugin.ConfigureRequest_GlobalConfig{TrustDomain: "example.org"},
 	})
-	s.RequireGRPCStatus(err, codes.Unknown, `k8s-sat: cluster "FOO" configuration must have at least one service account whitelisted`)
+	s.RequireGRPCStatus(err, codes.Unknown, `k8s-sat: cluster "FOO" configuration must have at least one service account allowed`)
 	s.Require().Nil(resp)
 
-	// cluster missing service account whitelist (token review validation config)
+	// cluster missing service account allow_list (token review validation config)
```
Comment (azdagron) [inline pkg/server/plugin/nodeattestor/k8s/sat/sat_test.go:344]:
```suggestion
	// cluster missing service account allow list (token review validation config)
```

File: conf/server/server_full.conf [hunk 7]
```diff
@@ -307,12 +307,12 @@ plugins {
...
     #         # clusters = {
     #             # "<arbitrary ID>" = {
-    #                 # service_account_whitelist: A list of service account names,
+    #                 # service_account_allow_list: A list of service account names,
     #                 # qualified by namespace (for example, "default:blog" or
     #                 # "production:web") to allow for node attestation. Attestation
     #                 # will be rejected for tokens bound to service accounts that
-    #                 # aren't in the whitelist.
-    #                 # service_account_whitelist = []
+    #                 # aren't in the allow_list.
```
Comment (azdagron) [inline conf/server/server_full.conf:314]:
```suggestion
    #                 # aren't in the allow list.
```

File: conf/server/server_full.conf [hunk 8]
```diff
@@ -345,12 +345,12 @@ plugins {
...
```
Comment (azdagron) [inline conf/server/server_full.conf:352]:
```suggestion
    #                 # aren't in the allow list.
```

File: doc/plugin_server_nodeattestor_gcp_iit.md [hunk 9]
```diff
@@ -4,13 +4,13 @@
 
 The `gcp_iit` plugin automatically attests instances using the [GCP Instance Identity Token](https://cloud.google.com/compute/docs/instances/verifying-instance-identity). It also allows an operator to use GCP Instance IDs when defining SPIFFE ID attestation policies.
 Agents attested by the gcp_iit attestor will be issued a SPIFFE ID like `spiffe://TRUST_DOMAIN/spire/agent/gcp_iit/PROJECT_ID/INSTANCE_ID`
-This plugin requires a whitelist of ProjectID from which nodes can be attested. This also means that you shouldn't run multiple trust domains from the same GCP project.
+This plugin requires an allow_list of ProjectID from which nodes can be attested. This also means that you shouldn't run multiple trust domains from the same GCP project.
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_gcp_iit.md:7]:
```suggestion
This plugin requires an allow list of ProjectID from which nodes can be attested. This also means that you shouldn't run multiple trust domains from the same GCP project.
```

File: doc/plugin_server_nodeattestor_k8s_psat.md [hunk 10]
```diff
@@ -25,7 +25,7 @@ Each cluster in the main configuration requires the following configuration:
 
 | Configuration | Description | Default                 |
 | ------------- | ----------- | ----------------------- |
-| `service_account_whitelist` | A list of service account names, qualified by namespace (for example, "default:blog" or "production:web") to allow for node attestation. Attestation will be rejected for tokens bound to service accounts that aren't in the whitelist. | |
+| `service_account_allow_list` | A list of service account names, qualified by namespace (for example, "default:blog" or "production:web") to allow for node attestation. Attestation will be rejected for tokens bound to service accounts that aren't in the allow_list. | |
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_k8s_psat.md:28]:
```suggestion
| `service_account_allow_list` | A list of service account names, qualified by namespace (for example, "default:blog" or "production:web") to allow for node attestation. Attestation will be rejected for tokens bound to service accounts that aren't in the allow list. | |
```

File: doc/plugin_server_nodeattestor_k8s_sat.md [hunk 11]
```diff
@@ -31,7 +31,7 @@ Each cluster in the main configuration requires the following configuration:
 
 | Configuration | Description | Default                 |
 | ------------- | ----------- | ----------------------- |
-| `service_account_whitelist` | A list of service account names, qualified by namespace (for example, "default:blog" or "production:web") to allow for node attestation. Attestation will be rejected for tokens bound to service accounts that aren't in the whitelist. | |
+| `service_account_allow_list` | A list of service account names, qualified by namespace (for example, "default:blog" or "production:web") to allow for node attestation. Attestation will be rejected for tokens bound to service accounts that aren't in the allow_list. | |
```
Comment (azdagron) [inline doc/plugin_server_nodeattestor_k8s_sat.md:34]:
```suggestion
| `service_account_allow_list` | A list of service account names, qualified by namespace (for example, "default:blog" or "production:web") to allow for node attestation. Attestation will be rejected for tokens bound to service accounts that aren't in the allow list. | |
```

**Review by azdagron — COMMENTED** [review]
Next stop: nitpick city :)

LGTM, other than some spots where we should either be using the configurable name (e.g. `projectid_allow_list`) or the conceptual name (e.g, `project ID allow list`) but instead have neither (e.g. `allow_list`)

**Review by azdagron — APPROVED** [review]
\o/

**Comment by evan2645** [comment]
Tagging the tracking issue https://github.com/spiffe/spire/issues/2003 for posterity

**Pull request merged**